cannot match err to none

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 0.2.2 6a54f3a
**OS:** Ubuntu Focal 20.04

**What did you do?**
```v
fn do_a_thing(i int) ?int {
	if i < 0 {
		return error("can't be negative")
	}
	if i == 0 {
		return none
	}
	return i
}

fn test_test() {
	i := 0
	if r := do_a_thing(i) {
		assert false
	} else {
		match err {
			none {}
			else {
				assert false
			}
		}
	}
}
```

**What did you expect to see?**
The err variable to be matchable to none or else, or other error types.
 
**What did you see instead?**
```
/tmp/v/_test.4487736817512997117.tmp.c:13852: error: invalid operand types for binary operation
/tmp/v/_test.4487736817512997117.tmp.c:13852: error: invalid aggregate type for register load
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -5332,6 +5332,8 @@ pub fn (mut c Checker) match_expr(mut node ast.MatchExpr) ast.Type {
 	cond_type_sym := c.table.get_type_symbol(cond_type)
 	if cond_type_sym.kind !in [.interface_, .sum_type] {
 		node.is_sum_type = false
+	} else {
+		node.is_sum_type = true
 	}
 	c.match_exprs(mut node, cond_type_sym)
 	c.expected_type = cond_type

--- a/vlib/v/tests/match_error_to_none_test.v
+++ b/vlib/v/tests/match_error_to_none_test.v
@@ -1,0 +1,25 @@
+fn do_a_thing(i int) ?int {
+	if i < 0 {
+		return error("can't be negative")
+	}
+	if i == 0 {
+		return none
+	}
+	return i
+}
+
+fn test_match_error_to_none() {
+	i := 0
+	if r := do_a_thing(i) {
+		println(r)
+	} else {
+		match err {
+			none {
+				assert true
+			}
+			else {
+				assert false
+			}
+		}
+	}
+}

